include/eve/detail/function/simd/arm/neon/slide_left.hpp

```diff
@@ -0,0 +1,67 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/bit_cast.hpp>
+#include <eve/conditional.hpp>
+
+namespace eve::detail
+{
+  template<real_scalar_value T, typename N, arm_abi ABI, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T,N,ABI> slide_left_ ( EVE_SUPPORTS(neon128_)
+                                            , wide<T,N,ABI> v, index_t<Shift>
+                                            ) noexcept
+  requires(Shift<=N::value)
+  {
+    using that_t  = wide<T,N,ABI>;
+
+    if constexpr(Shift == N::value)
+    {
+      return that_t{0};
+    }
+    else if constexpr(Shift == 0)
+    {
+      return v;
+    }
+    else
+    {
+      constexpr auto c = categorize<that_t>();
+      that_t z{0};
+
+      // Mask noises from smaller sized registers
+      if constexpr(N::value < expected_cardinal_v<T,arm_64_>)
+      {
+        wide<T,expected_cardinal_t<T,arm_64_>> w = v.storage();
+        w &= keep_first(N::value).mask(as(w)).bits();
+        v = w.storage();
+      }
```

Comment:
I think that you should mask after the shift.
I believe that as is it leaves the shifted values instead of 0s.

Reply:
I mask over the expected size, this means I mask values physically in the storage but not in the logical view.
Maybe it'll be simpler to do it the other way. Will try.

Reply:
Shift have been simplified